kodi_addon_checker/schema_validation.py

```diff
@@ -0,0 +1,114 @@
+    elif failed:
+        report.add(Record(PROBLEM, "Schema validation failed for the following points: %s " % ''.join(failed)))
+
+
+def _validation_checks(report: Report, parsed_xml, branch_name):
+    """Perform certain checks on the XML elements and extension point
+    """
+    metadatacount = 0
+    validcount = 1
+    failed = []
+    valid_points = {
+        'kodi.context.item': 'contextitem.xsd',
+        'kodi.game.controller': 'controller.xsd',
+        'kodi.resource.games': 'games.xsd',
+        'kodi.resource.images': 'images.xsd',
+        'kodi.resource.language': 'language.xsd',
+        'kodi.addon.metadata': 'metadata.xsd',
+        'kodi.resource.uisounds': 'uisound.xsd',
+        'xbmc.addon.metadata': 'metadata.xsd',
+        'xbmc.addon.repository': 'repository.xsd',
+        'xbmc.gui.skin': 'skin.xsd',
+        'xbmc.metadata.scraper.albums': 'scraper.xsd',
+        'xbmc.metadata.scraper.artists': 'scraper.xsd',
+        'xbmc.metadata.scraper.movies': 'scraper.xsd',
+        'xbmc.metadata.scraper.musicvideos': 'scraper.xsd',
+        'xbmc.metadata.scraper.tvshows': 'scraper.xsd',
+        'xbmc.metadata.scraper.library': 'scraper.xsd',
+        'xbmc.python.script': 'script.xsd',
+        'xbmc.python.lyrics': 'script.xsd',
+        'xbmc.python.weather': 'script.xsd',
+        'xbmc.python.library': 'script.xsd',
+        'xbmc.python.pluginsource': 'pluginsource.xsd',
+        'xbmc.python.module': 'script.xsd',
+        'xbmc.service': 'service.xsd',
+        'xbmc.subtitle.module': 'script.xsd',
+        'xbmc.ui.screensaver': 'script.xsd',
+        'xbmc.webinterface': 'webinterface.xsd',
+    }
+
+    for extension in parsed_xml.findall("extension"):
+        extension_point = extension.get("point")
+        schema_path = check_version(branch_name, valid_points[extension_point])
+
+        if extension_point in valid_points and schema_path:
+            if extension_point == "xbmc.addon.metadata" or extension_point == "kodi.addon.metadata":
+                metadatacount += 1
+
+            if not os.path.exists(schema_path):
+                report.add(Record(PROBLEM, "%s doesn't exist" % schema_path))
+                continue
+
+            if not _validate(extension, schema_path):
+                failed.append(extension_point)
+                validcount -= 1
+            parsed_xml.remove(extension)
+        else:
+            report.add(Record(PROBLEM, "%s is not a valid extension point" % extension_point))
+            validcount -= 1
+
+    return failed, metadatacount, validcount
+
+
+def _validate(xml, schemapath):
+    """Validates given XML file/element with a given schema
+
+    :xml: parsed xml element/file
+    :schemapath: path to a schema file
+    """
+    try:
+        schema = xmlschema.XMLSchema(schemapath)
+        schema.validate(xml)
+        return True
+    except xmlschema.XMLSchemaValidationError as schemaerror:
+        print(schemaerror)
+        return False
```

Comment:
replace the whole try/catch logic with the following to get all validation errors and not only the first one
```
schema = xmlschema.XMLSchema(schemapath)
for error in schema.iter_errors(xml):
    print(error)
return schema.is_valid(xml)
```